src/main/java/org/broadinstitute/hellbender/utils/collections/IntervalsSkipList.java

```diff
@@ -0,0 +1,130 @@
+package org.broadinstitute.hellbender.utils.collections;
+
+import com.google.common.collect.Lists;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+/**
+ * Holds many intervals, with an efficient operation to get
+ * intervals that overlap a given query interval.
+ *
+ * This version assumes that all the intervals lie on the same contig.
+ */
+public final class IntervalsSkipList<T extends Locatable> implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    // approx number of buckets we're aiming for.
+    private final int NUMBUCKETS = 1000;
+    // each bucket contains 2**SHIFT entries.
+    private final int SHIFT;
+    // reach: bucket# -> how far that bucket reaches.
+    // e.g. bucket 0 contains the first 2**SHIFT locatables. reach[0] is the max over their .getEnd()
+    //      reach[x] is the max over the .getEnd for that bucket and all the ones before it.
+    private int[] reach;
+
+    // input intervals, sorted by start location
+    private ArrayList<T> vs;
+
+    /**
+     * Creates an IntervalsSkipList that holds a copy of the given intervals, sorted
+     * and indexed.
+     *
+     * @param loc Locatables, not necessarily sorted. Will be iterated over exactly once.
+     */
+    public IntervalsSkipList(Iterable<T> loc) {
+        vs = Lists.newArrayList(loc);
+        int bSize = vs.size() / NUMBUCKETS;
+        if (bSize<32) bSize=32;
+        SHIFT = log2(bSize);
+        sortVariants();
+        buildIndex();
+    }
+
+    /**
+     * returns all the intervals that overlap with the query.
+     * You may modify the returned list.
+     */
+    public ArrayList<T> getOverlapping(SimpleInterval query) {
```

Comment:
yup, exactly. Ideally Locatable would have it, but since it doesn't, we have to do this.